src/RoomDirectoryModel.h

```diff
@@ -0,0 +1,88 @@
+// SPDX-FileCopyrightText: 2021 Nheko Contributors
+//
+// SPDX-License-Identifier: GPL-3.0-or-later
+
+#pragma once
+
+#include <QAbstractListModel>
+#include <QHash>
+#include <QString>
+#include <string>
+#include <vector>
+
+#include "MatrixClient.h"
+#include <mtx/responses/public_rooms.hpp>
+#include <mtxclient/http/errors.hpp>
+
+#include "Logging.h"
+
+namespace mtx::http {
+using RequestErr = const std::optional<mtx::http::ClientError> &;
+}
+namespace mtx::responses {
+struct PublicRooms;
+}
+
+class RoomDirectoryModel : public QAbstractListModel
+{
+        Q_OBJECT
+
+public:
+        explicit RoomDirectoryModel(QObject *parent = nullptr, const std::string &s = "");
+
+        enum Roles
+        {
+                Name = Qt::UserRole,
+                Id,
+                AvatarUrl,
+                Topic,
+                MemberCount,
+                Previewable
+        };
+        QHash<int, QByteArray> roleNames() const override;
+
+        QVariant data(const QModelIndex &index, int role) const override;
+
+        inline int rowCount(const QModelIndex &parent = QModelIndex()) const override
+        {
+                (void)parent;

```

Comment:
Nitpick: you could use `Q_UNUSED` instead, but it's not a big deal

Reply:
We use (void) in most places. Q_UNUSED is Qt specific and I personally don't think it is more readable. Alternatively there is also [[maybe_unused]]

Reply:
Yup I matched the convention to use void. [[maybe_unused]] would make Modern C++ fanatics happier but it seems a bit verbose for me in this context